src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -573,6 +589,12 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             }
             catch (Exception e)
             {
+                if (e is OperationCanceledException operationException && TimeoutFired(callerToken, timeoutTime))
```

Comment:
It looks like this could be an exception filter.

Reply:
If it's an exception filter, then you need two catch blocks; doesn't really help anything.

Reply:
I find the two catch blocks cleaner; is there a reason to avoid catch blocks when they have entirely exclusive bodies? perf concerns?

Reply:
Previously they weren't entirely discrete:
https://github.com/dotnet/runtime/pull/2281#discussion_r373209969
As they are now, it's really in the eye of the beholder.  The two blocks with an exception filter will add some expense when the exception occurs, but given that it's only in the exceptional case, it's not prohibitive.  Whichever you both decide is fine.

Reply:
@scalablecory Do you want me to change it back to the exception filters? I'd rather not do this since it will require rerunning all perf tests for Linux and Windows which takes a lot of time.

Reply:
My opinion is that this is fine as-is. If we think this is a problem later, you can change it.